### PR TITLE
Dynamic configuration of http headers

### PR DESCRIPTION
**Pull request by viniciussoares**
Allow http headers to be reconfigured through HTTPNetworkTransport.

This will probably resolve the following issues:

#37 
#140 
#131 (Never been merged)

---

/enhancement

- [x] feature
- [ ] blocking
- [ ] docs

### CODE REVIEW

**Comment by amol-c** [comment]
Can we get this PR merged as a temporary fix even if there are plan for longer term fix?

**Comment by viniciussoares** [comment]
@amol-c as a temporary workaround update your Podfile to use my repository.

```
pod 'Apollo',        git: 'https://github.com/viniciussoares/apollo-ios', branch: 'master'
pod 'Apollo/SQLite', git: 'https://github.com/viniciussoares/apollo-ios', branch: 'master'
```

File: Sources/Apollo/HTTPNetworkTransport.swift [hunk 1]
```diff
@@ -70,7 +79,11 @@ public class HTTPNetworkTransport: NetworkTransport {
   public func send<Operation>(operation: Operation, completionHandler: @escaping (_ response: GraphQLResponse<Operation>?, _ error: Error?) -> Void) -> Cancellable {
     var request = URLRequest(url: url)
     request.httpMethod = "POST"
-    
+
+    httpHeaders?.forEach { header in
+        request.addValue(header.value, forHTTPHeaderField: header.key)
```
Comment (rajkhatri) [inline Sources/Apollo/HTTPNetworkTransport.swift]: You have 2 extra spaces before `request` for consistent indentation

**Comment by brennantaylor** [comment]
Could we instead get a delegate? This would make per-request headers much easier (ie per request signing).

**Comment by martijnwalraven** [comment]
I agree with @brennantaylor that adding a property for headers won't be flexible enough to support the use cases that have come up. I added some more detail about this in https://github.com/apollographql/apollo-ios/pull/297#issuecomment-400535981.

**Comment by viniciussoares** [comment]
I agree :) Closed in favor of #297

**Comment by brennantaylor** [comment]
@viniciussoares Thanks, your PR inspired me to get off my butt and open one of my own :p

**Pull request closed without merging**